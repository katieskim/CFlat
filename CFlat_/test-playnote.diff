0a1,3
> tone: C-
> octave: 4
> rhythm: s.
