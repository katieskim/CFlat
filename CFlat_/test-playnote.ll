; ModuleID = 'CFlat'
source_filename = "CFlat"

%named_struct_note_t = type { i8*, i32, i8* }

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt.1 = private unnamed_addr constant [16 x i8] c"/%s/ /%d/ /%s/\0A\00"
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00"
@fmt.3 = private unnamed_addr constant [4 x i8] c"%s\0A\00"
@fmt.4 = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt.5 = private unnamed_addr constant [4 x i8] c"%s\0A\00"
@fmt.6 = private unnamed_addr constant [4 x i8] c"%s\0A\00"
@tone_ptr = private unnamed_addr constant [3 x i8] c"C-\00"
@rhythm_ptr = private unnamed_addr constant [2 x i8] c"s\00"

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

declare i32 @play_note(%named_struct_note_t*)

define i32 @main() {
entry:
  %n = alloca %named_struct_note_t
  store %named_struct_note_t { i8* getelementptr inbounds ([3 x i8], [3 x i8]* @tone_ptr, i32 0, i32 0), i32 3, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @rhythm_ptr, i32 0, i32 0) }, %named_struct_note_t* %n
  %play_note = call i32 @play_note(%named_struct_note_t* %n)
  ret i32 0
}
