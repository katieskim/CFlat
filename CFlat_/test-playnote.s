	.text
	.file	"CFlat"
	.globl	main                    # -- Begin function main
	.p2align	4, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# %bb.0:                                # %entry
	subq	$24, %rsp
	.cfi_def_cfa_offset 32
	leaq	.Lrhythm_ptr(%rip), %rax
	movq	%rax, 16(%rsp)
	leaq	.Ltone_ptr(%rip), %rax
	movq	%rax, (%rsp)
	movl	$4, 8(%rsp)
	movq	%rsp, %rdi
	callq	play_note@PLT
	xorl	%eax, %eax
	addq	$24, %rsp
	retq
.Lfunc_end0:
	.size	main, .Lfunc_end0-main
	.cfi_endproc
                                        # -- End function
	.type	.Lfmt,@object           # @fmt
	.section	.rodata.str1.1,"aMS",@progbits,1
.Lfmt:
	.asciz	"%d\n"
	.size	.Lfmt, 4

	.type	.Lfmt.1,@object         # @fmt.1
.Lfmt.1:
	.asciz	"/%s/ /%d/ /%s/\n"
	.size	.Lfmt.1, 16

	.type	.Lfmt.2,@object         # @fmt.2
.Lfmt.2:
	.asciz	"%g\n"
	.size	.Lfmt.2, 4

	.type	.Lfmt.3,@object         # @fmt.3
.Lfmt.3:
	.asciz	"%s\n"
	.size	.Lfmt.3, 4

	.type	.Lfmt.4,@object         # @fmt.4
.Lfmt.4:
	.asciz	"%d\n"
	.size	.Lfmt.4, 4

	.type	.Lfmt.5,@object         # @fmt.5
.Lfmt.5:
	.asciz	"%s\n"
	.size	.Lfmt.5, 4

	.type	.Lfmt.6,@object         # @fmt.6
.Lfmt.6:
	.asciz	"%s\n"
	.size	.Lfmt.6, 4

	.type	.Ltone_ptr,@object      # @tone_ptr
.Ltone_ptr:
	.asciz	"C-"
	.size	.Ltone_ptr, 3

	.type	.Lrhythm_ptr,@object    # @rhythm_ptr
.Lrhythm_ptr:
	.asciz	"s."
	.size	.Lrhythm_ptr, 3


	.section	".note.GNU-stack","",@progbits
