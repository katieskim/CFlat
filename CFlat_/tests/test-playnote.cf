int main()
{
	note n;
	n = ( /C-/ /3/ /s/ );
	playnote(n);
	
	return 0;
}
