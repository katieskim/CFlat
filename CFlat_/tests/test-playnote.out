tone: C-
octave: 4
rhythm: s.
